core/src/main/java/org/elasticsearch/index/query/AndQueryParser.java

```diff
@@ -72,21 +72,11 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 } else if (parseContext.isDeprecatedSetting(currentFieldName)) {
                     // skip
                 } else if (token == XContentParser.Token.START_ARRAY) {
-                    if ("filters".equals(currentFieldName)) {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
-                        }
-                    } else {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
+                    queriesFound = true;
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                        Query filter = parseContext.parseInnerFilter();
+                        if (filter != null) {
+                            queries.add(filter);
```

Comment:
I think we need to allow the notation without the `filters` element, since this is also the default example given in the docs. The `filters` element seems to have been used only when adding e.g. an additional `_cache` element. Since this query is deprecated on master anyway I'm not sure if we need to fix this leniency here.